docs/index.md

```diff
@@ -2636,36 +2636,121 @@ detailed information about the assertion.
-would be *delighted* to help you complete that process.
+### Quick tour of the metamath-lamp source code
+
+If you wish to try to contribute *code* to the metamath-lamp
+project itself, you'll need to learn more. Here are some hints.
 
-Thank you for your time.
+The metamath-lamp program is written in the
+[ReScript programming language](https://rescript-lang.org/).
+ReScript is a robustly typed language that
+compiles to efficient and human-readable JavaScript.
+ReScript is a programming language
+similar to [OCaml](https://en.wikipedia.org/wiki/OCaml);
+it is multi-paradigm but it encourages functional programming approaches.
+ReScript source files have the extension `.res`.
+
+For more about ReScript, see this
+[introduction](https://rescript-lang.org/docs/manual/latest/introduction)
+and
+[overview](https://rescript-lang.org/docs/manual/latest/overview).
+ReScript has a sound type system and its code has no null/undefined errors.
+By default you must specify the full scoped name of a function, e.g.,
+`mytext->Js_string2.split(":")`.
+
+Metamath-lamp
+uses the widely-used React library to implement its user interface.
+React comes with a robust API to React (and JavaScript).
+
+In discussions, metamath-lamp is often abbreviated as `mm-lamp`.
+
+To simplify things and not have to think about different possible scenarios,
+metamath-lamp tends to use the approach
+*stop if bad data is detected and let the user decide how to correct the data*.
+
+Within the code:
+
+* Filenames beginning with `MM_cmp_*.res` (meaning "component")
+  are very coupled with UI related things, e.g., React, the web browser DOM,
+  window global variables, and so on.
+  Usually one cmp file contains one React component, but
+  this is not a strict rule and there are a few exceptions.
+* Files beginning with `MM_wrk_*.res` (meaning "worker")
+  must not depend on UI related things, as these files could be (eventually)
+  used in a worker thread. Worker threads don't support many features
+  which are supported in the main thread.
+  If such a UI related feature appears in a worker thread it
+  usually leads to an error (compilation or runtime).
+  Where possible, put functionality in `MM_wrk_` files instead of
+  `MM_cmp_` files; this means they can work in worker file *and*
+  makes them much easier to test.
+* Where at all possible, please reuse existing functions unless there's
+  a strong reason to do otherwise, as this simplifies code review.
+  It's acknowledged that this is challenge when you are new to the code base.
+* Functional style is generally applied, e.g., `filter` and `map`.
+* Like several other languages, many functions return an
+  `Option` (which can be `Some('a)` or `None`) or a
+  `Result` (which can be `Ok('good)` or `Error('bad)`). These enable
+  compile-time checking that prevent null and undefined errors.
+  However, if you need it (such as for stopping loops early),
+  ReScript also supports exceptions.
```

Comment:
```
However, if you need it (such as for stopping loops early),
ReScript also supports exceptions.
```

Please, don't put this to the section describing mm-lamp code :) This may be misleading. I try to avoid exceptions as a mechanism of control flow. There are few such usages of exceptions, but they are in the code I wrote when I didn't have enough experience in ReScript. In the new code, exceptions are used to only terminate the program with some meaningful message.